sdcm/cluster_k8s.py

```diff
@@ -0,0 +1,416 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2020 ScyllaDB
+
+# pylint: disable=too-many-arguments
+
+import os
+import logging
+from typing import Optional, Union, List, Dict
+from textwrap import dedent
+from functools import cached_property
+
+from sdcm import cluster, cluster_gce, cluster_docker
+from sdcm.remote import LOCALRUNNER
+from sdcm.sct_config import sct_abs_path
+from sdcm.sct_events import TestFrameworkEvent
+from sdcm.utils.k8s import KubernetesCmdRunner, KubernetesOps
+from sdcm.utils.common import get_free_port, wait_for_port
+from sdcm.utils.docker_utils import ContainerManager
+from sdcm.utils.remote_logger import get_system_logging_thread
+
+
+KUBECTL_PROXY_PORT = 8001
+KUBECTL_PROXY_CONTAINER = "auto_ssh:kubectl_proxy"
+SCYLLA_OPERATOR_CONFIG = sct_abs_path("sdcm/k8s_configs/operator.yaml")
+SCYLLA_CLUSTER_CONFIG = sct_abs_path("sdcm/k8s_configs/cluster.yaml")
+IPTABLES_BIN = "iptables"
+
+LOGGER = logging.getLogger(__name__)
+
+
+class MinikubeOps:
+    @classmethod
+    def setup_minikube(cls, node: cluster.BaseNode, kubectl_version: str, minikube_version: str) -> None:
+        LOGGER.debug("Prepare %s to run Minikube cluster", node)
+        if node.distro.is_ubuntu18:
+            cls.setup_minikube_ubuntu18(node, kubectl_version, minikube_version)
```

Comment:
any reason why we don't start on ubuntu19 ? it doesn't have LTS yet ?

Reply:
Next LTS is Ubuntu 20.04 and we have no support for it in SCT yet.